extensions/scaffold/plugin.h

```diff
@@ -0,0 +1,43 @@
+#ifndef NULL_PLUGIN
+
+#include "proxy_wasm_intrinsics.h"
+
+#else
+
+#include "include/proxy-wasm/null_plugin.h"
+
+namespace proxy_wasm {
+namespace null_plugin {
+namespace scaffold {
+
+#endif
+
+// PluginRootContext is the root context for all streams processed by the
+// thread. It has the same lifetime as the worker thread and acts as target for
+// interactions that outlives individual stream, e.g. timer, async calls.
+class PluginRootContext : public RootContext {
+ public:
+  explicit PluginRootContext(uint32_t id, std::string_view root_id)
+      : RootContext(id, root_id) {}
+
+  bool onConfigure(size_t) override;
+};
+
+// Per-stream context.
+class PluginContext : public Context {
+ public:
+  explicit PluginContext(uint32_t id, RootContext* root) : Context(id, root) {}
```

Comment:
Should we add available callbacks here as comment? Same comment to the root context.

Reply:
I think it's fine to leave as-is. I am mostly interested in performance measurements so want to keep it minimal.

Reply:
I think a different one (or same one with different #defines) that defines all callbacks is also important. You can add it as another plying though.

Reply:
Yeah, but they are easily discoverable since methods must be overridden and have default definitions.